### PR TITLE
Optimizations for TaskHub task claiming

### LINKED ISSUE #121: Compute-side bottleneck when many compute services attempt to claim from a single `TaskHub`
First reported [here](https://github.com/openforcefield/alchemiscale/pull/120#issuecomment-1510511650) in #120.

We observe a bottleneck compute-side with task claiming when there are a small number of `TaskHub`s/`AlchemicalNetwork`s with available `Task`s, since the claiming code must currently lock the `TaskHub` while performing claiming. This linearizes requests for new `Task`s from a given `TaskHub` by compute services, causing a bit of starving compute.

This can be mitigated by setting the number of tasks claimed at a time on compute services higher. The problem will also be diminished when many `TaskHub`s have waiting tasks, as well as when tasks take much longer than the ~1min tasks I've been testing with.

Addressing this issue directly will require a more efficient claiming code path. Ideally we would be able to claim `Task`s from a `TaskHub` in a single Cypher query, but this may not be possible given our approach to claiming. Alternative to this, a rework of the claiming code that avoids the need to lock the `TaskHub` being claimed from may also yield higher aggregate performance.

### PR DESCRIPTION
**Pull request by ianmkenney**
This PR optimizes task claiming by putting more burden on Neo4j for priority sorting, removing the use of Subgraphs, intelligently stopping early, and generally sending fewer queries across the entire process. This closes #121 and should be reviewed after #242 is merged.

### CODE REVIEW

File: alchemiscale/storage/statestore.py [hunk 1]
```diff
@@ -96,15 +97,29 @@ def _select_task_from_taskpool(taskpool: Subgraph) -> Union[ScopedKey, None]:
     return chosen_one[0]
 
 
+def _select_tasks_from_taskpool(taskpool: List[Tuple[str, float]], count) -> List[str]:
```
Comment (dotsdl) [inline alchemiscale/storage/statestore.py:100]: Since we have this, can we delete `_select_task_from_taskpool` above?

Comment (dotsdl) [inline alchemiscale/storage/statestore.py:100]: Add a small docstring.

**Review by dotsdl — APPROVED** [review]
Looks great @ianmkenney! Fantastic work on this!

Just one note on removing dead code, and I added clarifying comments where I thought they would help us in the future if we have to revisit this.

**Comment by dotsdl** [comment]
Once you've made changes you think are appropriate, please merge!

File: alchemiscale/storage/statestore.py (hunk 1 above)
Comment (ianmkenney) [reply alchemiscale/storage/statestore.py]: Removed

Comment (ianmkenney) [reply alchemiscale/storage/statestore.py]: Added docstring

File: alchemiscale/storage/statestore.py [hunk 2]
```diff
@@ -136,14 +136,15 @@ def _generate_claim_query(
...
     WHERE NOT (t)<-[:CLAIMS]-(:ComputeServiceRegistration)
-    SET t.status = 'running'
 
     WITH t
 
     // create CLAIMS relationship with given compute service
     MATCH (csreg:ComputeServiceRegistration {{identifier: '{compute_service_id}'}})
     CREATE (t)<-[cl:CLAIMS {{claimed: localdatetime('{datetime.utcnow().isoformat()}')}}]-(csreg)
 
+    SET t.status = 'running'
```
Comment (ianmkenney) [inline alchemiscale/storage/statestore.py]: updating this to use `TastStatusEnum.running.value`

**Comment by ianmkenney** [comment]
Merging on CI pass

File: alchemiscale/storage/statestore.py (hunk 2 above)
Comment (dotsdl) [reply alchemiscale/storage/statestore.py]: Awesome, thank you!

**Pull request merged**